src/app/ReadClient.cpp

```diff
@@ -1271,6 +1272,15 @@ CHIP_ERROR ReadClient::DefaultResubscribePolicy(CHIP_ERROR aTerminationCause)
         return CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT;
     }
 
+    if (aTerminationCause.IsIMStatus() && ClusterStatusCode(aTerminationCause).GetStatus() == Status::InvalidAction)
+    {
+        // InvalidAction in most cases means some Interaction Model mismatch with the target device
+        // That's not something that could be fixed by retry
+        ChipLogError(DataManagement, "Stop resubscription due to unrecoverable error %" CHIP_ERROR_FORMAT,
```

Comment:
I disagree that this is an unrecoverable error.  We just don't know, based on the information we have whether it's recoverable....